src/main/java/uk/gov/beis/els/model/RatingClass.java

```diff
@@ -35,4 +40,20 @@ public String getPlusses() {
     return plusses;
   }
 
+  static {
+    Map<String, RatingClass> map = new HashMap<String, RatingClass>();
+    for (RatingClass instance : RatingClass.values()) {
+      map.put(instance.getDisplayValue().toLowerCase(), instance);
+    }
+    ENUM_MAP = Collections.unmodifiableMap(map);
+  }
+
+  public static RatingClass getEnum(String ratingClass) {
```

Comment:
I think this logic can be simplified a bit, e.g.
```
public static RatingClass getEnum(String ratingClass) {
    return Arrays.stream(RatingClass.values())
        .filter(e -> e.getDisplayValue().equalsIgnoreCase(ratingClass) || e.name().equals(ratingClass))
        .findFirst()
        .orElseThrow(() -> new RuntimeException("Can't resolve RatingClass from string '" + ratingClass + "'"));
  }
```
This way you don't need the separate static map, and its a bit easier to read in my opinion (due to the more semantic streams API). 
Same for the other enums

Reply:
done